frontend/pkg/frontend/ocm.go

```diff
@@ -107,6 +107,79 @@ func (f *Frontend) ConvertCStoHCPOpenShiftCluster(ctx context.Context, systemDat
 	return hcpcluster, nil
 }
 
+func (f *Frontend) ConvertCStoHCPOpenShiftClusterbySubscription(cluster *cmv1.Cluster) (*api.HCPOpenShiftCluster, error) {
```

Comment:
Potentially naive feedback, but we should use `ConvertCStoHCPOpenShiftCluster` instead - was there some issue with using it?

Reply:
Suggest just passing `&arm.SystemData{}` to `ConvertCStoHCPOpenShiftCluster` for now, and revisit this after https://issues.redhat.com/browse/ARO-8002 is resolved.